cluster/aws/cleanup-preset-vm.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# Copyright 2020 Authors of Arktos.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_VER=""
+KUBEMARK_MASTER=${KUBEMARK_MASTER:-false}
+
+# this scipt is to clean up arktos in the preset machine
+
+function set-kube-env() {
+  local kube_env_yaml="/etc/kubernetes/kube_env.yaml"
+
+  # kube-env has all the environment variables we care about, in a flat yaml format
+  eval "$(python -c '
+import pipes,sys,yaml
+
+for k,v in yaml.load(sys.stdin).iteritems():
+  print("""readonly {var}={value}""".format(var = k, value = pipes.quote(str(v))))
+  print("""export {var}""".format(var = k))
+  ' < """${kube_env_yaml}""")"
+}
+
+function stop-flannel-ds {
+  pushd /tmp
+  wget https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+  kubectl --kubeconfig=/home/ubuntu/.kube/config delete -f /tmp/kube-flannel.yml || true
+  rm -rf /tmp/kube-flannel.yml
+  popd
+}
+
+function stop-containers {
+  kubeadm reset -f
+  docker system prune -f
+  docker stop $(docker ps -aq) || true
+  docker rmi $(docker images -q)
+}
+
+function stop-services() {
+  systemctl stop kubelet
```

Comment:
I think you don't need stop kubelet if you use "kubeadm reset".

Reply:
Tried. This is necessary. Will get "'/usr/bin/kubelet': Text file busy" without it.

Reply:
Did you do "kubeadm reset" on minions?

Reply:
Yes. When clean up vms, I don't differentiate master and minion.

Reply:
Then that is not right. kubeadm supposes to stop any kube related services and containers. After kubeadm reset, kubelet should already been stopped and uninstalled from services. If it is not, our kubeadm reset has issue. It should be fixed, not using a workaround.

Reply:
After did more test, I think the issue I hit is transient. I removed it.